mailpoet/assets/js/src/notices/invalid-mss-key-notice.tsx

```diff
@@ -4,24 +4,56 @@ import ReactStringReplace from 'react-string-replace';
 
 type Props = {
   mssKeyInvalid: boolean;
+  premiumKeyValid: boolean;
   subscribersCount: number;
 };
 
-function InvalidMssKeyNotice({ mssKeyInvalid, subscribersCount }: Props) {
+function InvalidMssKeyNotice({
+  mssKeyInvalid,
+  premiumKeyValid,
+  subscribersCount,
+}: Props) {
   if (!mssKeyInvalid) return null;
   return (
     <Notice type="error" timeout={false} closable={false} renderInPlace>
       <h3>{MailPoet.I18n.t('allSendingPausedHeader')}</h3>
       <p>
-        {ReactStringReplace(
-          MailPoet.I18n.t('allSendingPausedBody'),
-          /\[link\](.*?)\[\/link\]/g,
-          (match) => (
-            <a href="?page=mailpoet-settings#premium" key="check-sending">
-              {match}
-            </a>
-          ),
-        )}
+        {premiumKeyValid &&
+          ReactStringReplace(
+            MailPoet.I18n.t('allSendingPausedPremiumValidBody'),
+            /\[link\](.*?)\[\/link\]/g,
+            (match, index) => {
+              // first link tag
+              if (index === 1) {
+                return (
+                  <a
+                    href={`https://account.mailpoet.com?s=${subscribersCount}`}
```

Comment:
Would it make sense to use an upgrade link instead? `Kindly upgrade to a [link]MailPoet sending plan[/link]`?

Reply:
Yes, this makes sense. What UTM params should this link have? Same as on the Premium upgrade page?

Reply:
Mostly. The ref should change though to something that specifies the origin like `invalid_mss_key_notice` or something else.